src/NuGet.Core/NuGet.Configuration/Settings/SettingFactory.cs

```diff
@@ -99,8 +100,24 @@ internal static SettingBase Parse(XNode node, SettingsFile origin)
         internal static IEnumerable<T> ParseChildren<T>(XElement xElement, SettingsFile origin, bool canBeCleared) where T : SettingElement
         {
             var children = new List<T>();
-
-            var descendants = xElement.Elements().Select(d => Parse(d, origin)).OfType<T>().Distinct();
+            IEnumerable<T> descendants;
+            if (xElement.Name.LocalName.Equals(ConfigurationConstants.PackageSourceMapping, StringComparison.OrdinalIgnoreCase))
```

Comment:
This code builds a list of duplicates first and if none are found it returns the list of the decedents again.  Is it worth making this more performant by looping through the decedents once while keeping track of duplicates?  Even that would always require a hashtable.  I just think this is a pretty hot path and would hate to see it allocate or be slow when _most of the time_ people don't have duplicates.  Is it worth optimizing this logic more?

Reply:
I think this feedback is still valid, but not yet addressed. 

Distinct tends to use a set already I think. Should be easy to run a test to compare the performance.